sos/plugins/kubernetes.py

```diff
@@ -35,7 +35,22 @@ class kubernetes(Plugin, RedHatPlugin):
     def check_is_master(self):
         if any([
             path.exists("/var/run/kubernetes/apiserver.key"),
-            path.exists("/etc/origin/master/master-config.yaml")
+            path.exists("/etc/origin/master/master-config.yaml"),
+            path.exists("/root/cdk/server.key")
+        ]):
+            return True
+        return False
+
+    def check_is_ubuntu(self):
```

Comment:
The Plugin class already has this kind of OS-detection functionality built-in with the tagging mechanic.

This means we can just move `kube_cmd` to a class attr, and then use something like this for Ubuntu:

```python
class UbuntuKubernetes(kubernetes, UbuntuPlugin):

    files = ("/root/cdk/server.key", )

    def setup():
        self.kube_cmd = "snap run kubectl "
        super(UbuntuKubernetes, self).setup()
```

After that we'd just need to either transform `self.kube_cmd` back into `kube_cmd` in `Kubernetes.setup()` or modify all references of kube_cmd to be `self.kube_cmd` (probably best).